src/connection.cpp

```diff
@@ -182,32 +220,25 @@ void Connection::parsePacket(const boost::system::error_code& error)
 	if (error) {
 		close(FORCE_CLOSE);
 		return;
-	} else if (closed) {
+	} else if (connectionState == CONNECTION_STATE_DISCONNECTED) {
 		return;
 	}
 
-	//Check packet checksum
-	uint32_t checksum;
-	int32_t len = msg.getLength() - msg.getBufferPosition() - NetworkMessage::CHECKSUM_LENGTH;
-	if (len > 0) {
-		checksum = adlerChecksum(msg.getBuffer() + msg.getBufferPosition() + NetworkMessage::CHECKSUM_LENGTH, len);
-	} else {
-		checksum = 0;
-	}
-
-	uint32_t recvChecksum = msg.get<uint32_t>();
-	if (recvChecksum != checksum) {
-		// it might not have been the checksum, step back
-		msg.skipBytes(-NetworkMessage::CHECKSUM_LENGTH);
-	}
+	// Read potential checksum bytes
+	uint32_t recvPacket = msg.get<uint32_t>();
 
 	if (!receivedFirst) {
-		// First message received
 		receivedFirst = true;
 
 		if (!protocol) {
+			// Skip deprecarted checksum bytes (with clients that aren't using it in mind)
+			uint16_t len = msg.getLength();
+			if (len < 280 && len != 151) {
```

Comment:
I would like to know what are these numbers xD

Reply:
151 is packet length for 8.6 and 9.7 (and probably all the clients in the middle)
clients that use checksums have packet lengths larger than 280 although I'm not sure how was that figured out